tiny_dnn/core/framework/tensor.h

```diff
@@ -415,12 +400,15 @@ class Tensor {
    * A tensor holds data in C-style nD array, i.e row-major order:
    * the rightmost index “varies the fastest”.
    */
-  std::array<size_t, kDimensions> shape_;
+  std::vector<size_t> shape_;
 
   /* Offset from the beginning of TensorStorage */
   size_t offset_;
   size_t size_;
 
+  /* Offset in case of non-contiguous memory */
+  size_t stride_;
```

Comment:
offset_ can be negative in case of bottom up data. so `ptrdiff_t` might be good to use.